src/lib/project-detection/index.ts

```diff
@@ -44,34 +44,57 @@ export const identifyProjectSlug = async ({
 
 /**
  * Get the pipeline number from the URL
- * @param {string} url - eg: https://app.circleci.com/pipelines/gh/organization/project/2/workflows/abc123de-f456-78gh-90ij-klmnopqrstuv
- * @returns {string} pipeline number - eg: 2
+ * @param {string} url - CircleCI pipeline URL
+ * @returns {number} The pipeline number
+ * @example
+ * // Standard pipeline URL
+ * getPipelineNumberFromURL('https://app.circleci.com/pipelines/gh/organization/project/2/workflows/abc123de-f456-78gh-90ij-klmnopqrstuv')
+ * // returns 2
+ *
+ * @example
+ * // Pipeline URL with complex project path
+ * getPipelineNumberFromURL('https://app.circleci.com/pipelines/circleci/GM1mbrQEWnNbzLKEnotDo4/5gh9pgQgohHwicwomY5nYQ/123/workflows/abc123de-f456-78gh-90ij-klmnopqrstuv')
+ * // returns 123
  */
-export const getPipelineNumberFromURL = (url: string) => {
+export const getPipelineNumberFromURL = (url: string): number => {
   const parts = url.split('/');
-  return parts[7];
-};
+  const pipelineIndex = parts.indexOf('pipelines');
+  if (pipelineIndex === -1) {
+    throw new Error('Invalid CircleCI URL format');
+  }
+  const pipelineNumber = parts[pipelineIndex + 4];
 
-/**
- * Get the project slug from the URL
- * @param {string} url - eg: https://app.circleci.com/pipelines/gh/organization/project/2/workflows/abc123de-f456-78gh-90ij-klmnopqrstuv
- * @returns {string} project slug - eg: gh/organization/project
- */
-export const getProjectSlugFromURL = (url: string) => {
-  const parts = url.split('/');
-  return `${parts[4]}/${parts[5]}/${parts[6]}`;
+  if (!pipelineNumber) {
+    throw new Error('Unable to extract pipeline number from URL');
+  }
+
+  const parsedNumber = Number(pipelineNumber);
+  if (isNaN(parsedNumber)) {
+    throw new Error('Pipeline number in URL is not a valid number');
+  }
+
+  return parsedNumber;
 };
 
 /**
  * Get the project slug from the URL
- * @param {string} url - eg: https://app.circleci.com/pipelines/gh/organization/project
+ * @param {string} url - CircleCI pipeline or project URL
  * @returns {string} project slug - eg: gh/organization/project
+ * @example
+ * // Pipeline URL with workflow
+ * getProjectSlugFromURL('https://app.circleci.com/pipelines/gh/organization/project/2/workflows/abc123de-f456-78gh-90ij-klmnopqrstuv')
+ * // returns 'gh/organization/project'
+ *
+ * @example
+ * // Simple project URL
+ * getProjectSlugFromURL('https://app.circleci.com/pipelines/gh/organization/project')
+ * // returns 'gh/organization/project'
  */
-export const getProjectSlugFromProjectURL = (url: string) => {
+export const getProjectSlugFromURL = (url: string) => {
```

Comment:
made `getProjectSlugFromURL` a bit more robust by looking for the `pipelines` section of the url and using it as a starting point.

we can get the projectSlug from multiple types of urls, so this better describes what it is doing. and i removed the extra `getProjectSlugFromProjectURL` that effectively did the same thing